From 7486b5f52872d548736c4529080c7cb5abafc10e Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Fri, 24 Jun 2022 15:23:32 -0700
Subject: [PATCH 2/2] Revert "mm: Add an assertion that PG_private and
 folio->private are in sync"

This reverts commit c462f63731e0fd1a50f4df385a2529ba88b40af8.

This triggers with zram.

Link: https://lore.kernel.org/YrYzhB7aDNIBz%2FuV@dev-arch.thelio-3990X/
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 mm/filemap.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/mm/filemap.c b/mm/filemap.c
index 529386e9d1e7..8ccb868c3d95 100644
--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -1535,9 +1535,6 @@ void folio_unlock(struct folio *folio)
 	BUILD_BUG_ON(PG_waiters != 7);
 	BUILD_BUG_ON(PG_locked > 7);
 	VM_BUG_ON_FOLIO(!folio_test_locked(folio), folio);
-	VM_BUG_ON_FOLIO(!folio_test_swapbacked(folio) &&
-			(folio_test_private(folio) ==
-			 !folio_get_private(folio)), folio);
 	if (clear_bit_unlock_is_negative_byte(PG_locked, folio_flags(folio, 0)))
 		folio_wake_bit(folio, PG_locked);
 }
-- 
2.36.1

